lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4742,6 +4742,10 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
 /// \returns true on error, false on success.
 static bool typeCheckDerivativeAttr(ASTContext &Ctx, Decl *D,
                                     DerivativeAttr *attr) {
+  if (D->isInvalid()) {
+    return true;
+  }
+  
```

Comment:
We would want to check whether differentiable programming is enabled below before doing this.